src/main/java/htsjdk/variant/vcf/VCFCodec.java

```diff
@@ -72,91 +65,68 @@
-            }
-            else {
-                throw new TribbleException.InvalidHeader("We never saw the required CHROM header line (starting with one #) for the input VCF file");
-            }
+    public boolean canDecodeVersion(final VCFHeaderVersion targetHeaderVersion) {
+        return targetHeaderVersion == VCFHeaderVersion.VCF4_0 ||
+                targetHeaderVersion == VCFHeaderVersion.VCF4_1 ||
+                targetHeaderVersion == VCFHeaderVersion.VCF4_2;
+    }
 
-        }
-        throw new TribbleException.InvalidHeader("We never saw the required CHROM header line (starting with one #) for the input VCF file");
+    @Override
+    public boolean canDecode(final String potentialInput) {
+        // TODO: this will succeed on 4.3 files since it only looks as far as ..."##fileformat=VCFv4"
+        return canDecodeFile(potentialInput, VCF4_MAGIC_HEADER);
     }
 
     /**
-     * parse the filter string, first checking to see if we already have parsed it in a previous attempt
-     *
-     * @param filterString the string to parse
-     * @return a set of the filters applied or null if filters were not applied to the record (e.g. as per the missing value in a VCF)
+     * Handle reporting of duplicate filter IDs
+     * @param duplicateFilterMessage
      */
     @Override
-    protected List<String> parseFilters(final String filterString) {
-        // null for unfiltered
-        if ( filterString.equals(VCFConstants.UNFILTERED) )
-            return null;
-
-        if ( filterString.equals(VCFConstants.PASSES_FILTERS_v4) )
-            return Collections.emptyList();
-        if ( filterString.equals(VCFConstants.PASSES_FILTERS_v3) )
-            generateException(VCFConstants.PASSES_FILTERS_v3 + " is an invalid filter name in vcf4", lineNo);
-        if (filterString.isEmpty())
-            generateException("The VCF specification requires a valid filter status: filter was " + filterString, lineNo);
-
-        // do we have the filter string cached?
-        if ( filterHash.containsKey(filterString) )
-            return filterHash.get(filterString);
-
-        // empty set for passes filters
-        final List<String> fFields = new LinkedList<String>();
-        // otherwise we have to parse and cache the value
-        if ( !filterString.contains(VCFConstants.FILTER_CODE_SEPARATOR) )
-            fFields.add(filterString);
-        else
-            fFields.addAll(Arrays.asList(filterString.split(VCFConstants.FILTER_CODE_SEPARATOR)));
+    protected void reportDuplicateFilterIDs(final String duplicateFilterMessage) {
+        // older versions of htsjdk have been silently dropping these for a while, but we can at least warn
+        if (VCFUtils.getVerboseVCFLogging()) {
+            logger.warn(duplicateFilterMessage);
+        }
+    }
 
-        filterHash.put(filterString, Collections.unmodifiableList(fFields));
+    public void reportDuplicateInfoKeyValue(final String key, final String infoLine) {}
```

Comment:
Excellent question. Historically, htsjdk reads multiple VCF versions, but always writes only the current version. That always worked because we could always coerce old versions to the current version. But not all existing VCFs can be auto-converted to V4.3 in a non-destructive way since 4.3 is somewhat more restrictive. So we have to make a decision about to integrate v4.3. i.e., do we require opt-in to 4.3 ? IIRC, this method is a no-op based on the idea that we would implement a new VCF43Codec that WOULD report these, but thats part of the philosophical discussion that we need to resolve for this PR. I think there multiple places in this PR that will be affected by how we resolve that issue.